### PR TITLE
Better collisions

### PR DESCRIPTION
**Pull request by BenjaminCall**
Working pretty good

### CODE REVIEW

File: scripts/state/state.js [hunk 1]
```diff
@@ -62,35 +62,59 @@ class State {
     }
   }
 
-  collisions(data) {
+  collisions(allBullets, allShips) {
```
Comment (BenHSmith13) [inline scripts/state/state.js]: this should go in its own file

Comment (BenHSmith13) [reply scripts/state/state.js]: probably

File: scripts/state/state.js [hunk 2]
```diff
@@ -62,35 +62,59 @@ class State {
...
     let bulletsVis = [];
-    _.forEach(data.bullets, (bullet)=>{
-      if(bullet.visible){bulletsVis.push(bullet)}
+    _.forEach(allBullets, (bullet)=>{
+      if(bullet.visible && bullet.owner === 'player'){bulletsVis.push(bullet)}
     })
     let shipsVis = [];
```
Comment (BenHSmith13) [inline scripts/state/state.js]: why decrare new arrays?

Comment (BenjaminCall) [reply scripts/state/state.js]: It makes for cleaner checks and a lot less iterating

File: scripts/state/collision/collision.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+class Collision {
+
+  collisions(allBullets, allShips, sounds) {
+    let bulletsVis = [];
```
Comment (BenHSmith13) [inline scripts/state/collision/collision.js:4]: why declare new arrays?

Comment (BenjaminCall) [reply scripts/state/collision/collision.js:4]: makes it so I don't iterate over a ton of non visible bullets, lessens the amount of work alot

Comment (BenHSmith13) [reply scripts/state/collision/collision.js:4]: ok, most of the bullets will be visible (like 80% to 90%) with the system so we really don't gain much from this, and this is just iterating over them an extra time. but its fine.

**Review by BenHSmith13 — APPROVED** [review]

File: scripts/state/collision/collision.js (hunk 3 above)
Comment (BenjaminCall) [reply scripts/state/collision/collision.js:4]: well with 50 bullets initialized and only like 10 or so bullets on the screen..

**Pull request merged**